packages/dropdown/src/js/index.ts

```diff
@@ -85,168 +98,218 @@ const sortDropdownProps = ({
   }
 })
 
+// TODO: move to react folder? Uses React APIs.
 export const useDropdown = (
   props: UseDropdownProps,
   forwardedRef: Ref<HTMLButtonElement>
 ) => {
   const { hook, ...rest } = sortDropdownProps(props)
   const [isOpen, setOpen] = useState(false)
+  const inputId = useMemo(() => uniqueId('dropdown-input-'), [])
+  const menuId = useMemo(() => uniqueId('dropdown-menu-'), [])
 
-  const itemMatchingValue = useMemo(() => {
-    return findMatchingActionMenuItem(hook.menu, hook.value)
-  }, [hook.menu, hook.value])
-
-  const [selectedLabel, setSelectedLabel] = useState(
-    itemMatchingValue ? itemMatchingValue.props.children : null
+  const items = useMemo(() => parseMenuChildren(menuId, hook.menu), [
+    menuId,
+    hook.menu
+  ])
+  const itemMatchingValueIndex = items.findIndex(
+    item => item.value === hook.value || item.label === hook.value
   )
+  const itemMatchingValue = items[itemMatchingValueIndex]
+  const [activeIndex, setActiveIndex] = useState(
+    itemMatchingValueIndex > -1 ? itemMatchingValueIndex : 0
+  )
+
+  const [selectedLabel, setSelectedLabel] = useState(itemMatchingValue?.label)
   const [selectedValue, setSelectedValue] = useState(hook.value)
 
   useEffect(() => {
-    setSelectedLabel(
-      itemMatchingValue ? itemMatchingValue.props.children : null
-    )
-    setSelectedValue(hook.value)
+    const newLabel = itemMatchingValue?.label
+    const newValue = hook.value
+    setSelectedLabel(newLabel)
+    setSelectedValue(newValue)
   }, [itemMatchingValue, hook.value])
 
-  function handleToggleOpen(evt) {
+  useEffect(() => {
+    function handleEscape(evt) {
+      if (evt.key === 'Escape') {
+        setOpen(false)
+        setActiveIndex(itemMatchingValueIndex > -1 ? itemMatchingValueIndex : 0)
+        buttonRef.current?.focus()
+      }
+    }
+
+    if (canUseDOM() && isOpen) {
+      document.addEventListener('keydown', handleEscape, false)
+
+      return () => {
+        document.removeEventListener('keydown', handleEscape, false)
+      }
+    }
+  }, [itemMatchingValueIndex, isOpen])
+
+  function handleButtonEvent(evt: React.MouseEvent | React.KeyboardEvent) {
+    if (
+      evt.type === 'click' ||
+      (evt.type === 'keydown' &&
+        'key' in evt &&
+        (evt.key === ' ' || evt.key === 'Enter' || evt.key === 'ArrowDown'))
+    ) {
+      evt.preventDefault()
+      evt.stopPropagation()
+
+      const newOpen = !isOpen
+      setOpen(newOpen)
+      if (newOpen && inputRef.current) {
+        inputRef.current.focus()
+      }
+      if (typeof hook.onClick === 'function') hook.onClick(evt)
+    }
+  }
+
+  function handleInputKeyDown(evt: React.KeyboardEvent) {
     evt.preventDefault()
     evt.stopPropagation()
-    setOpen(!isOpen)
-    if (typeof hook.onClick === 'function') hook.onClick(evt)
-  }
 
-  function handleKeyDown(evt: React.KeyboardEvent) {
-    if (evt.key === 'ArrowDown') {
-      setOpen(true)
+    if (isOpen) {
```

Comment:
Wondering if we just need the ActionMenu markup and styles and not the key handlers on it?

Reply:
Also we seem to have lost keyboard events on the menu. I can't key through them anymore after running clean build and then storybook.

Reply:
Not sure what you mean by this:

> Wondering if we just need the ActionMenu markup and styles and not the key handlers on it?

Mostly because we need keyboard support, and that would seem to remove it.

On your 2nd comment:

> seem to have lost keyboard events on the menu

What keyboard events have you lost specifically?  Keyboard support has been a large focus of this refactor, so if I'm messing that up, I'm way off the mark.  Should be supported:

- Tab - focus button
- Enter/Space/ArrowDown - activate button to open menu 
- ArrowDown/ArrowUp - move to items in menu
- Enter/Space - select item 
- Escape - close menu

I have just freshly rebuilt and run test and storybook.  Auto and manual tests are looking good for me.  Please confirm and further specify the problem you're seeing.